h11/_headers.py

```diff
@@ -56,6 +56,7 @@
 # Maybe a dict-of-lists would be better?
 
 _content_length_re = re.compile(br"[0-9]+")
+_header_value_re = re.compile(br"[\x21-\xff]+(?:[ \t]+[\x21-\xff]+)*|^$")
```

Comment:
That was my first choice, but we get a bit of an import mess because _readers, _state, _and _event all import * from each other.

_state is dependent upon _connection being imported before it, which is handled in __init__.py. However, _events is imported first in __init__ which causes the problem. _events imports _headers, which will then import from _readers which imports everything from _state. _state fails because it doesn't have what it needs from _connection, which hasn't been imported yet.

If the definitions are going to be nicely reusable, they'll likely need to be moved into either a utils or definitions file.